via/models/_mixins.py

```diff
@@ -0,0 +1,20 @@
+from datetime import datetime
+
+from sqlalchemy import func
+from sqlalchemy.orm import Mapped, MappedAsDataclass, mapped_column
+
+
+class CreatedUpdatedMixin(MappedAsDataclass):
+    created: Mapped[datetime] = mapped_column(
+        init=False,
+        repr=False,
+        server_default=func.now(),  # pylint:disable=not-callable
+        sort_order=-10,
+    )
+    updated: Mapped[datetime] = mapped_column(
+        init=False,
+        repr=False,
+        server_default=func.now(),  # pylint:disable=not-callable
+        onupdate=func.now(),  # pylint:disable=not-callable
+        sort_order=-10,
+    )
```

Comment:
This is based on the [`TimestampMixin` example from the SQLAlchemy docs](https://docs.sqlalchemy.org/en/20/orm/declarative_mixins.html#mixing-in-columns) and also taking some ideas from [our own `CreatedUpdatedMixin` in LMS](https://github.com/hypothesis/lms/blob/e9f4491ffdc07cf586dc7a36303dd1de165aa0cd/lms/models/_mixins/created_updated.py#L4-L11) (using `server_default`, `onupdate` and `func.now()` for the timestamp fields so they get created and updated automatically).

I removed the timestamps from the instance's string representation (`repr=False`) because they're quite noisy when printed out.

I also removed the timestamps from the model's `__init__()` method (`init=False`) because I didn't want users to have to provide `created` and `updated` arguments every time you create a `Transcript`. You can add a `default=func.now()` (rather than `server_default`) so that `created` and `updated` become optional `__init__()` but because these columns are in a mixin class this then results in an error when SQLAlchemy tries to create the Python dataclass:

    sqlalchemy.exc.InvalidRequestError: Python dataclasses error encountered when creating dataclass for 'Transcript': TypeError("non-default argument 'video_id' follows default argument"). Please refer to Python dataclasses documentation for additional information. (Background on this error at: https://sqlalche.me/e/20/dcte)

A downside of the current implementation is that if you actually want to create a `Transcript` and provide your own `created` and `updated` columns you can't pass `created` and `updated` as arguments to `Transcript.__init__()`. Instead you'll have to do this:

```python
transcript = Transcript(...)
transcript.created = ...
transcript.updated = ...
```

I think this is the best compromise. In practice you never do want to set `created` and `updated` yourself anyway

Reply:
The `sort_order=-10` is to make sure that the `created` and `updated` columns come first in the SQL schema. Otherwise I think we could have problems with the order of the columns if any new columns are added to `Transcript` in future